LogMonitor/src/LogMonitor/EtwMonitor.cpp

```diff
@@ -291,52 +306,6 @@ void WINAPI EtwMonitor::OnEventRecordTramp(
     }
 }
 
-///
-/// Function automatically called by ProcessTrace when an ETW event arrives. If
-/// it returns false, the ProcessTrace will stop receiving events.
-///
-BOOL WINAPI
-EtwMonitor::StaticBufferEventCallback(
-    _In_ PEVENT_TRACE_LOGFILE Buffer
-    )
-{
-    try
-    {
-        EtwMonitor* etwMon = static_cast<EtwMonitor*>(Buffer->Context);
-        return etwMon->BufferEventCallback(Buffer);
-    }
-    catch (std::exception& ex)
-    {
-        logWriter.TraceError(
-            Utility::FormatString(L"Failed to process ETW event callback. %S", ex.what()).c_str()
-        );
-        return FALSE;
-    }
-    catch (...)
-    {
-        logWriter.TraceError(
-            Utility::FormatString(L"Failed to process ETW event callback. Unknown error occurred").c_str()
-        );
-        return FALSE;
-    }
-}
-
-///
-/// Returns false if the m_stopFlag is set true.
-///
-BOOL WINAPI
-EtwMonitor::BufferEventCallback(
-    _In_ PEVENT_TRACE_LOGFILE Buffer

```

Comment:
Why do you have to remove it? If you return false from here, then ProcessTrace may return early.